examples/mnist_wgan.py

```diff
@@ -0,0 +1,336 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Train an Wasserstein Generative Adversarial Network (WGAN)
+on the MNIST dataset.
+See https://arxiv.org/abs/1610.09585 for more details about ACGAN.
+See https://arxiv.org/abs/1701.07875 for more details about WACGAN.
+
+You should start to see reasonable images after ~3 epochs.
+According to the paper, the performance is highly related
+to the discriminator loss.
+You should use a GPU, as the convolution-heavy operations
+are very slow on the CPU.
+Prefer the TensorFlow backend if you plan on iterating,
+as the compilation time can be a blocker using Theano.
+
+Timings:
+
+Hardware           | Backend | Time / Epoch
+-------------------------------------------
+ CPU               | TF      | 3 hrs
+ Titan X (maxwell) | TF      | 4 min
+ Titan X (maxwell) | TH      | 7 min
+
+Consult https://github.com/bobchennan/Wasserstein-GAN-Keras for
+more information and example output
+The original ACGAN implementation can be found in
+https://github.com/lukedeo/keras-acgan
+More tricks to train GAN can be found in https://github.com/soumith/ganhacks
+"""
+from __future__ import print_function
+
+from collections import defaultdict
+try:
+    import cPickle as pickle
+except ImportError:
+    import pickle
+from PIL import Image
+
+from six.moves import range
+
+import keras.backend as K
+from keras.datasets import mnist
+from keras.layers import Input, Dense, Reshape, Flatten
+from keras.layers import Dropout, Embedding, merge
+from keras.layers.advanced_activations import LeakyReLU
+from keras.layers.convolutional import UpSampling2D, Convolution2D
+from keras.layers.noise import GaussianNoise
+from keras.models import Sequential, Model
+from keras.optimizers import Adam, SGD
+from keras.backend.common import _EPSILON
+from keras.utils.generic_utils import Progbar
+import numpy as np
+
+np.random.seed(1331)
+
+K.set_image_dim_ordering('th')
+
+
+def modified_binary_crossentropy(target, output):
```

Comment:
I think you should just call this `wasserstein_loss` since there is really no relationship to binary cross-entropy any longer.